tests/e2e/utils/common_k8s_utils.go

```diff
@@ -0,0 +1,399 @@
+package utils
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	routev1 "github.com/openshift/api/route/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/remotecommand"
+	"log"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"strings"
+)
+
+type ProxyPodParameters struct {
+	KubeClient    *kubernetes.Clientset
+	KubeConfig    *rest.Config
+	Namespace     string
+	PodName       string
+	ContainerName string
+}
+
+type RequestParameters struct {
+	ProxyPodParams *ProxyPodParameters // Only required when using proxy
+	RequestMethod  *HTTPMethod
+	URL            string
+	Payload        *string // Not always required
+}
+
+// GetInternalServiceEndpointURL constructs the internal service endpoint URI
+// for a service in a Kubernetes cluster.
+//
+// Parameters:
+//   - namespace: The namespace of the service.
+//   - serviceName: The name of the service.
+//   - servicePort: (Optional) The port number of the service. If not provided,
+//     the default port 8000 is used.
+//
+// Returns:
+//   - string: The constructed internal service endpoint URI.
+func GetInternalServiceEndpointURL(namespace, serviceName string, servicePort ...int) string {
+	port := 8000
+	if len(servicePort) > 0 {
+		port = servicePort[0]
+	}
+
+	return fmt.Sprintf("http://%s.%s.svc.cluster.local:%d", serviceName, namespace, port)
+}
+
+// GetPodByLabel retrieves a pod in the specified namespace based on the given label selector.
+// It uses the provided Kubernetes client to interact with the Kubernetes API.
+//
+// Parameters:
+//
+//	kubeClient: A pointer to the Kubernetes client (*kubernetes.Clientset).
+//	namespace: The namespace in which to search for the pod.
+//	labelSelector: The label selector to filter pods.
+//
+// Returns:
+//
+//	(*corev1.Pod, error): A pointer to the first pod matching the label selector, or an error if any.
+//
+// Example:
+//
+//	pod, err := GetPodByLabel(kubeClient, "mongo-persistent", "curl-tool=true")
+//	if err != nil {
+//	    log.Println(err)
+//	}
+//	log.Printf("Pod found: %s\n", pod.Name)
+func GetPodByLabel(kubeClient *kubernetes.Clientset, namespace, labelSelector string) (*corev1.Pod, error) {
```

Comment:
```suggestion
func GetFirstPodByLabel(kubeClient *kubernetes.Clientset, namespace, labelSelector string) (*corev1.Pod, error) {
```
This func return the first one pod by label.. I think this makes it clearer.

Reply:
Will change.